src/utils/file_manager.py

```diff
@@ -0,0 +1,137 @@
+            result["resume"] = resume_file
+
+        return result
+
+    @staticmethod
+    def read_secrets(secrets_path: Path) -> dict:
+        """
+        Reads and validates the secrets file (YAML format) and ensures required keys are present.
+
+        Args:
+            secrets_path (Path): Path to the secrets YAML file.
+
+        Returns:
+            dict: Dictionary containing the secrets.
+
+        Raises:
+            FileNotFoundError: If the file does not exist.
+            ValueError: If the file is not a valid YAML file or required keys are missing.
+        """
+        if not secrets_path.exists():
+            raise FileNotFoundError(f"Secrets file not found: {secrets_path}")
+
+        try:
+            with open(secrets_path, "r", encoding="utf-8") as file:
+                secrets_data = yaml.safe_load(file)
+
+                # Ensure the secrets contain the required key for API
+                required_keys = ["llm_api_key"]
+                missing_keys = [key for key in required_keys if key not in secrets_data]
+                if missing_keys:
+                    raise ValueError(
+                        f"Missing required keys in secrets file: {', '.join(missing_keys)}"
+                    )
+
+                # Handle optional keys with default behavior
+                secrets_data["email"] = secrets_data.get("email", None)
+                secrets_data["password"] = secrets_data.get("password", None)
+
+                return secrets_data
+        except yaml.YAMLError as exc:
+            raise ValueError(f"Error parsing secrets file {secrets_path}: {exc}")
+
+    @staticmethod
+    def write_to_file(job, file_name, output_file_directory, reason=None, applicants_count=None):
+        logger.debug(f"Writing job application result to file: {file_name}")
+        pdf_path = Path(job.resume_path).resolve().as_uri() if job.resume_path else None
+        data = {
+            "company": job.company or "N/A",
+            "job_title": job.title or "N/A",
+            "link": job.link or "N/A",
+            "job_recruiter": job.recruiter_link or "N/A",
+            "job_location": job.location or "N/A",
+            "pdf_path": pdf_path or "N/A",
+            "reason": reason or "N/A"
+        }
+
+        if applicants_count is not None:
+            data["applicants_count"] = applicants_count
+
+        file_path = output_file_directory / f"{file_name}.json"
+        temp_file_path = file_path.with_suffix('.tmp')
+
+        try:
+            if not file_path.exists():
+                with open(temp_file_path, 'w', encoding='utf-8') as f:
+                    json.dump([data], f, indent=4)
+                temp_file_path.rename(file_path)
+                logger.debug(f"Job data written to new file: {file_path}")
+            else:
+                with open(file_path, 'r+', encoding='utf-8') as f:
+                    try:
+                        existing_data = json.load(f)
```

Comment:
refacor this. don't load file each write call

Reply:
let file manager manage those variables - JSON object and file object.

if you wish you move json object to context/state manager.

Reply:
The write_to_file method was refactored to minimize repeated file loading. The FileManager now manages open files and caches JSON data to avoid redundant reads on every write operation. This improves performance by reducing I/O overhead.